mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -160,14 +189,60 @@ public void update(CRUDOperationContext ctx,
                 }
                 docIndex++;
             }
+            // if we have any remaining items to update
+            if (numUpdating > 0) {
+                try {
+                    executeAndLogBulkErrors(bwo);
+                    numFailed += (int) docUpdateErrors.stream().map(e -> e.getIndex()).distinct().count();
```

Comment:
A better way of doing this is to concentrate the logic to a single place:
there is already a function that determines what's an error and what's not.
Count the failed docs there.

On Tue, May 31, 2016 at 9:01 AM, Brandon Vulaj <notifications@github.com>
wrote:

> In
> mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/275#discussion_r65199310>
> :
>
> > @@ -160,14 +189,60 @@ public void update(CRUDOperationContext ctx,
> >                  }
> >                  docIndex++;
> >              }
> > +            // if we have any remaining items to update
> > +            if (numUpdating > 0) {
> > +                try {
> > +                    executeAndLogBulkErrors(bwo);
> > +                    numFailed += (int) docUpdateErrors.stream().map(e -> e.getIndex()).distinct().count();
>
> If we want to ignore that case, we can simply just use the size of the
> error and update attempt lists to set numUpdated and numFailed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/275/files/2dafe4fde5cba55c66a0c86f8a3e10d8a285ef2f#r65199310>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADgDDaBq7ktNHmAtgL-QlOTuilFfOdpRks5qHE00gaJpZM4Ihlpv>
> .
>